src/pages/iou/request/step/IOURequestStepDistance.js

```diff
@@ -112,6 +108,30 @@ function IOURequestStepDistance({
         setShouldShowAtLeastTwoDifferentWaypointsError(false);
     }, [atLeastTwoDifferentWaypointsError, duplicateWaypointsError, hasRouteError, isLoading, isLoadingRoute, nonEmptyWaypointsCount, transaction]);
 
+    useEffect(() => {
+        if (isCreatingNewRequest) {
+            return () => {};
+        }
+        // This effect runs when the component is mounted and unmounted. It's purpose is to be able to properly
+        // discard changes if the user cancels out of making any changes. This is accomplished by backing up the
+        // original transaction, letting the user modify the current transaction, and then if the user ever
+        // cancels out of the modal without saving changes, the original transaction is restored from the backup.
+
+        // On mount, create the backup transaction.
+        setTransactionBackup(transaction);
+
+        return () => {
+            // If the user cancels out of the modal without without saving changes, then the original transaction
+            // needs to be restored from the backup so that all changes are removed.
+            if (transactionWasSaved.current || _.isEqual(transaction, transactionBackup)) {
+                return;
+            }
+            TransactionEdit.restoreOriginalTransactionFromBackup(lodashGet(transaction, 'transactionID', ''), transactionBackup);
+            setTransactionBackup({});
```

Comment:
Just FYI, the backup pattern could potentially restore an outdated data, for example:
1. Create a distance request
2. Edit the distance request. At this point, the transaction data will contains a pending data (amount 0, etc.)
3. Before the request succeeds, you open the edit distance page again which means the transaction backup now contains the pending data.
4. Edit the distance waypoints
5. After the first edit request completes, you close the page without saving and it will set the current transaction data with the backup (outdated) data.

Reply:
@bernhardoj Thanks for your concern. It seems this problem exist since we applied the backup transaction. IMO, it is a tiny case and hard to reproduce. But do you have any suggestions for this problem?

Reply:
I prefer saving any edit to ONYXKEYS.COLLECTION.TRANSACTION_DRAFT and save it to ONYXKEYS.COLLECTION.TRANSACTION when the user presses the save button.

I see that currently the data is saved directly to ONYXKEYS.COLLECTION.TRANSACTION.

Not sure though whether it will save your regression or not.

Reply:
Same idea to you, but it seems a bit risky 😄

Reply:
I was unsure about the transaction "backup" too, but it's true that the old code also did something similar. I wouldn't change much here unless it fixes actual regressions 🙁

The process doesn't reward improving the code if it introduces regression risks. Maybe intentionally, I'm not sure.

Reply:
I wouldn't either 😄